highlighters/main/main-highlighter.zsh

```diff
@@ -339,7 +339,22 @@ _zsh_highlight_highlighter_main_paint()
       (( start_pos += offset ))
       (( end_pos = start_pos + $#arg ))
     else
-      integer offset=$(((len-start_pos)-${#${proc_buf##([[:space:]]|\\[[:space:]])#}}))
+      # The line was:
+      #
+      # integer offset=$(((len-start_pos)-${#${proc_buf##([[:space:]]|\\[[:space:]])#}}))
+      #
+      # - len-start_pos is length of current proc_buf; basically: initial length minus where
+      #   we are, and proc_buf is chopped to the "where we are" (compare the "previous value
+      #   of start_pos" below, and the len-(start_pos-offset) = len-start_pos+offset)
+      # - what's after main minus sign is: lenghth of proc_buf without spaces at the beginning
+      # - so what the line actually did, was computing lenght of the spaces!
+      # - this can be done via (#b) flag, like below
+      if [[ "$proc_buf" = (#b)(#s)(([[:space:]]|\\[[:space:]])##)* ]]; then
+          # The first, outer parenthesis
+          offset="${#match[1]}"
```

Comment:
Need to declare `match` local.

Need to declare `offset` local.  (Not doing so breaks `make check`.)